### DISCUSSION TITLE
embed-resources returns a libs folder

### DISCUSSION THREAD

**Discussion by dmacoritto** [quarto-dev/quarto-cli#3436] [Q&A] [answered]
Hello everyone,

I am trying to switch from RMarkdown to Quarto and I have run into a challenge with the embed-resources YAML option. I would like to output an HTML document which is a standalone document with no dependencies.

Below is a screenshot of the YAML options for my document.

![image](https://user-images.githubusercontent.com/64702803/203508396-69b6caad-5c76-4fcc-be75-35f636169b31.png)

My issue is that, despite having included the "embed-resources: true" in the YAML, I get an external *_files folder that includes the required libraries to run the document (see below).  My charts are indeed embedded in my HTML document but not my libraries. Is there a way to get a complete self-contained document with no dependencies to a *_files folder ?

![image](https://user-images.githubusercontent.com/64702803/203509169-6714d2a8-77aa-420f-93bc-7936b9df5bcd.png)

Thank you very much for your help.

Darko

**Comment by cscheid** [comment]
Can you share the quarto version you're running and a .qmd that causes this problem?

**Reply by dmacoritto** [reply]
Hello @cscheid ,

The quarto version I am using is 1.2.269.

Please find a reproducible example there : 

https://github.com/dmacoritto/Productivity-Tools/blob/master/example.qmd

regards

**Reply by cscheid** [reply]
Ok, the issue here is that quarto uses `pandoc`'s old `self-contained` syntax. We convert to embed-resources, but apparently don't use embed-resources ourselves. This is a bug. In the meantime, use `self-contained`, and we'll fix in 1.3.

**Reply by dmacoritto** [reply]
I have tried changing `embed-resources: true` by `self-contained:true` but I get the exact same behaviour. A *_files with the libraries is generated and required to run the html document.

**Reply by dragonstyle** [reply, marked as answer]
I think the issue boils down to include `keep-md` in the front matter. Because you're keeping markdown, we are leaving the supporting  files on disk, since the markdown may contain references to those files. The HTML file that you're generating should be (and in my testing is) completely standalone with reference to the `*_files` directory.

If you remove the `keep-md` option, the supporting files directory should be properly cleaned up after render.

**Reply by dmacoritto** [reply]
You are right, the issue come from the option `keep-md: true`. When I remove it, the *_files directory disappears. Are you planning to generate a completely standalone markdown file in the future as well ?
Thank you for your reply and help.
